v3/src/models/document/shared-model-document-manager.ts

```diff
@@ -80,6 +83,24 @@ export class SharedModelDocumentManager implements ISharedModelDocumentManager {
     return this.document.addSharedModel(sharedModel)
   }
 
+  // TODO: Need to make this usable in CLUE
+  removeSharedModel(dataSetId: string) {
+    if (!this.document || !dataSetId) {
+      console.warn("removeSharedModel has no document or data set ID. this will have no effect")
+      return
+    }
+    const sharedDatasets = this.document.getSharedModelsByType<typeof SharedDataSet>("SharedDataSet")
+    const sharedMetadatas = this.document.getSharedModelsByType<typeof SharedCaseMetadata>("SharedCaseMetadata")
+    const dataSetToDeleteIndex = sharedDatasets.map(model => model.dataSet.id).indexOf(dataSetId)
+    const metadataToDeleteIndex = sharedMetadatas.map(model => model.data?.id).indexOf(dataSetId)
+    const sharedModelToDelete = sharedDatasets[dataSetToDeleteIndex]
+    const sharedCaseMetadataToDelete = sharedMetadatas[metadataToDeleteIndex]
+    // remove table associated with data set from document
+    const tileToRemove = this.getSharedModelTiles(sharedModelToDelete)
+                              .find(tile => tile.content.type === kCaseTableTileType)
```

Comment:
In theory, there could be multiple case table linked to this shared model, so I think this should be:
```suggestion
    const tilesToRemove = this.getSharedModelTiles(sharedModelToDelete)
                              .filter(tile => tile.content.type === kCaseTableTileType)
```
and then you'd pass the resulting `tilesToRemove` to the `removeSharedModelsAndTiles` function.